sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -219,16 +219,13 @@ private void checkNoMoreMatches(final List<PTransformOverride> overrides) {
           @Override
           public CompositeBehavior enterCompositeTransform(Node node) {
             if (!node.isRootNode()) {
-              for (PTransformOverride override : overrides) {
-                if (override.getMatcher().matches(node.toAppliedPTransform(getPipeline()))) {
-                  matched.put(node, override);
-                }
-              }
+              checkForMatches(node);
             }
-            if (!matched.containsKey(node)) {
+            if (matched.containsKey(node)) {
```

Comment:
nit: maybe a return value instead of checking the map

Reply:
The `PipelineVisitor` interface requires the `enterCompositeTransform()` method to return a `CompositeBehavior`, so I must return that. (Did I understand your comment correctly?)

Reply:
I think I will need to write some tests and I am working on it now.

Reply:
I run ValidatesRunner tests and the result is green.